File: common-lib/src/main/java/ar/edu/utn/frba/arbiter/common/enums/RiskBand.java
package ar.edu.utn.frba.arbiter.common.enums;

/**
 * Fraud/risk band of a claim's support score, ordered from least to most risk. The score is a
 * support signal for the analyst only: it never decides and never feeds the LLM prompt. The
 * score-to-band thresholds live in each insurer's scoring configuration, not here.
 */
public enum RiskBand {
    LOW,
    MEDIUM,
    HIGH,
    CRITICAL
}

File: arbiter-frontend/src/app/core/models/risk-band.ts
// Mirrors common-lib's RiskBand enum, ordered from lowest to highest risk.
export type RiskBand = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';

export const RISK_BANDS: RiskBand[] = ['LOW', 'MEDIUM', 'HIGH', 'CRITICAL'];

const LABELS: Record<RiskBand, string> = {
  LOW: 'Bajo',
  MEDIUM: 'Medio',
  HIGH: 'Alto',
  CRITICAL: 'Crítico',
};

export function riskBandLabel(band: RiskBand): string {
  return LABELS[band];
}
